src/mbgl/renderer/image_manager.cpp

```diff
@@ -39,6 +39,14 @@ void ImageManager::removeImage(const std::string& id) {
 
     auto it = patterns.find(id);
     if (it != patterns.end()) {
+        // Clear pattern from the atlas image.
+        const uint32_t w = it->second.bin->w;
+        const uint32_t h = it->second.bin->h;
+        const uint32_t x = it->second.bin->x;
+        const uint32_t y = it->second.bin->y;
+        PremultipliedImage blank { Size { w, h } };
+        PremultipliedImage::copy(blank, atlasImage, { 0, 0 }, { x, y }, { w, h });
```

Comment:
Let's add `PremultipliedImage::clear(img, { x, y }, { w, h })` so we don't have to allocate a blank image.